servers/src/main/java/tachyon/worker/block/BlockLockManager.java

```diff
@@ -70,8 +70,8 @@ public BlockLockManager(BlockMetadataManager metaManager) {
    * @throws IOException
    */
   public long lockBlock(long userId, long blockId, BlockLockType blockLockType) throws IOException {
-    // TODO: generate real hashValue on blockID.
-    int hashValue = (int) (blockId % (long) NUM_LOCKS);
+    // Knuth's multiplicative method: multiple by a number same order of hash size
+    int hashValue = (int) (Integer.toString((int) blockId).hashCode() * 997 % (long) NUM_LOCKS);
```

Comment:
check hash functions in guava here? 
http://docs.guava-libraries.googlecode.com/git-history/master/javadoc/com/google/common/hash/Hashing.html
I assume we don't need heavy-weight hash functions but choices like CRC might be sufficient.